rest/komand_rest/connection/connection.py

```diff
@@ -1,31 +1,101 @@
 import komand
-from .schema import ConnectionSchema
+from .schema import ConnectionSchema, Input
 from komand_rest.util.util import Common
+from komand.exceptions import PluginException
 # Custom imports below
-import base64
+from requests.auth import HTTPDigestAuth, HTTPBasicAuth
 
 
 class Connection(komand.Connection):
+    CUSTOM_SECRET_INPUT = "CUSTOM_SECRET_INPUT"
 
     def __init__(self):
         super(self.__class__, self).__init__(input=ConnectionSchema())
+        self.username = None
+        self.password = None
+        self.api_key = None
+        self.ssl_verify = None
+        self.base_url = None
+        self.default_headers = None
+        self.authentication_type = None
+        self.auth = None
 
-    def connect(self, params):
+    def connect(self, params={}):
         self.logger.info("Connect: Configuring REST details")
-        base_url = params.get("base_url")
-        default_headers = params.get("default_headers", {})
-        username = params.get('basic_auth_credentials', {}).get("username")
-        password = params.get("basic_auth_credentials", {}).get('password')
+        base_url = params.get(Input.BASE_URL)
+        default_headers = params.get(Input.DEFAULT_HEADERS, {})
+        self.authentication_type = params.get(Input.AUTHENTICATION_TYPE)
+        username = None
+        password = None
+        secret_key = None
```

Comment:
Do you want to use the `self.` versions instead? JW because they're assigned None values in the init of the class.